contents/security/Https.md

```diff
@@ -1,59 +1,62 @@
+### SHA-1 Collision Attack
+In the implementation of https, the CA does not sign the digital certificate directly, rather it signs the fixed length [hashed digest](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of the digital certificate for efficiency. This introduces the possibility of two different digital certificate with the same hashed digest. Thus, if the attackers manage to forge a fake digital certificate with the same hashed digest of another valid digital certificate, the browser would trust the attackers server and all the servers signed by the attackers. This loop hole occurs with SHA-1 hashing algorithm and SHA-1 is no longer used in https after 2016. In 2017, Google has announced an algorithm to forge a duplicated SHA-1 hash and the report is [here](https://github.com/se-edu/learningresources/pull/18/files). A more detailed explanation of this problem is found [here](https://www.sott.net/article/275524-Why-HTTPS-and-SSL-are-not-as-secure-as-you-think).
 
-2. During the hand shake process of https, client and server send each other in plain text the https standard they support and the most secure standard is chosen to be used. However, this can be exploited for https downgrade by a man in the middle attack. The man in the middle could send forged hand shake message to both server and client to indicate the maximum security supported is only 512 bits Diffie Hellman and trick server and client to encrypt with 512 bits Diffie Hellman. At the moment, a 512 bits Diffie Hellman algorithm could be brute forced with sufficient resources. A more detailed description could be found [here](https://weakdh.org/)
+### Weak Diffie-Hellman Attack
+During the [cipher suites](https://en.wikipedia.org/wiki/Cipher_suite) process of https, client and server send each other in plain text the https standard they support and the most secure standard is chosen to be used. However, this can be exploited for https downgrade by a man in the middle attack. The man in the middle could send forged hand shake message to both server and client to indicate the maximum security supported is only 512 bits Diffie Hellman and trick server and client to encrypt with 512 bits Diffie Hellman. At the moment, without knowing the keys, 512 bits Diffie Hellman algorithm could be decrypted with sufficient resources. A more detailed description could be found [here](https://weakdh.org/)
```

Comment:
Add a short explanation/link explaining man in the middle attack? Or does that link already explain it?